src/features/lireFlow/components/TextDisplay.tsx

```diff
@@ -1,11 +1,17 @@
 import "../layout/textDisplayStyle.css";
 import React, { useState } from "react";
 import WordHighlighter from "./WordHighlighter";
+import TextControls from "./TextControls";
 
 const TextDisplay = ({ fileContent }: { fileContent: string }) => {
-  const paragraphs: string[] = fileContent.split("\n");
+  const [speed, setSpeed] = useState<number>(1);
+  const paragraphs: string[] = fileContent
+    .split("\n")
+    .filter((paragraph: string) => paragraph.trim() !== "")
+    .map((paragraph: string) => paragraph.trim());

```

Comment:
You are using the Trim operation twice. It is also good to use useMemo for this case. If you use useMemo, you will repeat this operation only when the fileContent changes. This avoids unnecessary rerendering.

```suggestion
  const paragraphs: string[] = useMemo(() => {
    return fileContent
      .split("\n")
      .map((paragraph: string) => paragraph.trim())
      .filter((trimmedParagraph: string) => trimmedParagraph);
  }, [fileContent]);

```